contrib/charts/cert-manager/templates/letsencrypt_cluster_issuer.yaml

```diff
@@ -0,0 +1,25 @@
+{{- if and .Release.IsUpgrade .Values.letsencryptEmail }}
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt-production
+spec:
+  acme:
+    server: https://acme-v01.api.letsencrypt.org/directory
+    email: {{ .Values.letsencryptEmail }}
+    privateKeySecretRef:
+      name: letsencrypt-production-cluster-key-pair
+    http01: {}
```

Comment:
Note: we don't allow configuring dns providers via the Helm chart.

cc @ahmetb what do you think of this? In a previous comment on this PR, I noted that I *really* don't want us to start exposing this kind of config via Helm. We will basically be duplicating our API surface, which introduces a second layer of version skew/incompatibility that we have to manage. IMO, not being able to fully configure the Issuer/ClusterIssuer really limits the usefulness of this pull request. I imagine a lot of users will first use the default provisioned ones, and then potentially manually edit that created resource, and in future when they upgrade helm, all their manual changes will be blown away and they'll be left feeling confused (and their TLS will be broken).

Thinking about this scenario pushes me to once again think we *shouldn't* be doing this for users, and should once again force them to learn a bit more up front. It could definitely mislead users:

(a) how cert-manager works (they may be left thinking it is more similar to kube-lego than it is)
(b) what features cert-manager supports (how do I use DNS validation??)
(c) what they can and cannot modify. It blurs the boundary between 'deployment' configuration and 'runtime' configuration imo.

Reply:
IMO, we could have this chart manages all Issuers + ClusterIssuers.
Just having the values as an array, and we loop, and boom!

I think this is what people would expect (well, this is what I expect).

Should it be the scope of that PR, let me know.

The default yaml is too verbose, and heml provides a convenient way to strip this down to the minumum needed to describe our CRDs.